rtt/transports/mqueue/MQSendRecv.cpp

```diff
@@ -75,11 +76,14 @@ void MQSendRecv::setupStream(base::DataSourceBase::shared_ptr ds, base::PortInte
     marshaller_cookie = mtransport.createCookie();
     mis_sender = is_sender;
 
-    std::stringstream namestr;
-    namestr << '/' << port->getInterface()->getOwner()->getName() << '.' << port->getName() << '.' << this << '@' << getpid();
-
     if (policy.name_id.empty())
-        policy.name_id = namestr.str();
+    {
+        std::stringstream name_stream;
+        name_stream << port->getInterface()->getOwner()->getName() << '.' << port->getName() << '.' << this << '@' << getpid();
```

Comment:
Good to know ... I for one always assumes that this is the case.

As you say not really related to this PR (I just copied the existing code). However, as the goal here is to create a globally unique name, maybe it would be best to throw if there is no interface ? (Forcing the user to provide a name_id in this case)